gt/main.c

```diff
@@ -671,6 +642,26 @@ decap_and_fill_eth(struct rte_mbuf *m, struct gt_config *gt_conf,
 		return -1;
 	}
 
+	if (!is_neighbor) {
+		struct rte_ether_hdr *eth_hdr =
+			rte_pktmbuf_mtod(m, struct rte_ether_hdr *);
+		struct rte_ether_hdr *raw_eth = pkt_info->l2_hdr;
+
+		rte_ether_addr_copy(&raw_eth->s_addr, &eth_hdr->d_addr);
+		rte_ether_addr_copy(&raw_eth->d_addr, &eth_hdr->s_addr);
+		m->l2_len = gt_conf->net->front.l2_len_out;
+
+		if (iface->vlan_insert) {
+			fill_vlan_hdr(eth_hdr, iface->vlan_tag_be,
```

Comment:
1. Replace `gt_conf->net->front.` with `iface->`.
2. Variable `iface` is being used, but it is not defined in `decap_and_fill_eth()`. I understand that testing can be impossible, but please compile the code before submitting it.
3. Once variable `iface` is defined, search for other locations where it can be used in `decap_and_fill_eth()`.

Reply:
Thanks @AltraMayor ! Good catch!

2. Surprisingly, I did compiled it on my virtual machine and it passed.